pygmt/helpers/decorators.py

```diff
@@ -760,14 +767,23 @@ def new_module(*args, **kwargs):
                 if fmt in separators and is_nonstr_iter(value):
                     for index, item in enumerate(value):
                         if " " in str(item):
-                            # Check if there is a space " " when converting
-                            # a pandas.Timestamp/xr.DataArray to a string.
-                            # If so, use np.datetime_as_string instead.
-                            # Convert datetime-like item to ISO 8601
-                            # string format like YYYY-MM-DDThh:mm:ss.ffffff.
-                            value[index] = np.datetime_as_string(
-                                np.asarray(item, dtype=np.datetime64)
-                            )
+                            # Check if there is a space " " in the item, which
+                            # is typically present in objects such as
+                            # np.timedelta64, pd.Timestamp, or xr.DataArray.
+                            # If so, convert the item to a numerical or string
+                            # type that is understood by GMT as follows:
+                            if getattr(
+                                getattr(item, "dtype", ""), "name", ""
+                            ).startswith("timedelta"):
+                                # A np.timedelta64 item is cast to integer
+                                value[index] = item.astype("int")
```

Comment:
This double getattr to obtain `item.dtype.name` isn't very nice, wondering if there's a good alternative.